style_test.go

```diff
@@ -70,40 +98,64 @@ func TestStyleCustomRender(t *testing.T) {
 	}{
 		{
 			r.NewStyle().Foreground(Color("#5A56E0")),
-			"\x1b[38;2;89;86;224mhello\x1b[0m",
+			"\x1b[38;2;89;86;224mab c\x1b[0m",
 		},
 		{
 			r.NewStyle().Foreground(AdaptiveColor{Light: "#fffe12", Dark: "#5A56E0"}),
-			"\x1b[38;2;255;254;18mhello\x1b[0m",
+			"\x1b[38;2;255;254;18mab c\x1b[0m",
 		},
 		{
 			r.NewStyle().Bold(true),
-			"\x1b[1mhello\x1b[0m",
+			"\x1b[1mab c\x1b[0m",
 		},
 		{
 			r.NewStyle().Italic(true),
-			"\x1b[3mhello\x1b[0m",
+			"\x1b[3mab c\x1b[0m",
 		},
 		{
 			r.NewStyle().Underline(true),
-			"\x1b[4;4mh\x1b[0m\x1b[4;4me\x1b[0m\x1b[4;4ml\x1b[0m\x1b[4;4ml\x1b[0m\x1b[4;4mo\x1b[0m",
+			"\x1b[4;4ma\x1b[0m\x1b[4;4mb\x1b[0m\x1b[4m \x1b[0m\x1b[4;4mc\x1b[0m",
 		},
+        {
+            r.NewStyle().Underline(true).UnderlineSpaces(true),
+            "\x1b[4;4ma\x1b[0m\x1b[4;4mb\x1b[0m\x1b[4m \x1b[0m\x1b[4;4mc\x1b[0m",
+        },
+        {
+            r.NewStyle().Underline(true).UnderlineSpaces(false),
+            "\x1b[4;4ma\x1b[0m\x1b[4;4mb\x1b[0m \x1b[4;4mc\x1b[0m",
+        },
+        {
+            r.NewStyle().UnderlineSpaces(true),
+            "ab\x1b[4m \x1b[0mc",
+        },
 		{
-			r.NewStyle().Blink(true),
-			"\x1b[5mhello\x1b[0m",
+			r.NewStyle().Strikethrough(true),
+			"\x1b[9ma\x1b[0m\x1b[9mb\x1b[0m\x1b[9m \x1b[0m\x1b[9mc\x1b[0m",
```

Comment:
I would leave the existing test cases and put the new ones in a new TestUnderlineSpaces and StrikethroughSpaces cases

Reply:
Blink test case is still there, its just moved to the bottom

Reply:
Right, but your changes will taint the diff and blame on these lines 🙂

Reply:
i made TestUnderline and TestStrikethrough should i keep the underline tests from before(in TestStyleRender) or remove them?

Reply:
Let's keep them